### PR TITLE
feat(transfer-of-machine-ownership): aosh ondelete

### PR DESCRIPTION
**Pull request by sigruntg**
# On delete

Attach a link to issue if relevant

## What

Adding on delete functionality if owner deletes while in review

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Introduced functionality to generate email notifications for rejected applications for machine ownership transfers.
	- Added a new method to delete applications and notify relevant parties via email and SMS.
	- Implemented a new API endpoint for deleting machine owner change requests.
	- Added a new data provider for mockable payment catalog API.

- **Bug Fixes**
	- Enhanced error handling for email generation to ensure all necessary recipient details are present.

- **Documentation**
	- Updated API specifications to reflect new endpoints and parameters.

- **Chores**
	- Added mockable API configurations for testing purposes.

### CODE REVIEW

File: libs/application/template-api-modules/src/lib/modules/templates/aosh/transfer-of-machine-ownership/transfer-of-machine-ownership.service.ts [hunk 1]
```diff
@@ -270,6 +270,80 @@ export class TransferOfMachineOwnershipTemplateService extends BaseTemplateApiSe
...
+    application,
+    auth,
+  }: TemplateApiModuleActionProps): Promise<void> {
+    // 1. Delete charge so that the seller gets reimburshed
+    const chargeId = getPaymentIdFromExternalData(application)
+    if (chargeId) {
+      await this.chargeFjsV2ClientService.deleteCharge(chargeId)
+    }
+
+    console.log('DELETE HAPPENED')
```
Comment (berglindoma13) [inline libs/application/template-api-modules/src/lib/modules/templates/aosh/transfer-of-machine-ownership/transfer-of-machine-ownership.service.ts:283]: remove console log

Comment (berglindoma13) [inline libs/application/template-api-modules/src/lib/modules/templates/aosh/transfer-of-machine-ownership/transfer-of-machine-ownership.service.ts:283]: Remove console log

**Review by berglindoma13 — APPROVED** [review]
bara eitt lítið console log annars flott

**Review by disaerna — APPROVED** [review]
Code owned files 👍🏼

**Pull request merged**